tests/system/Database/Live/PreparedQueryTest.php

```diff
@@ -42,7 +43,11 @@ protected function tearDown(): void
         parent::tearDown();
 
         if ($this->query !== null) {
-            $this->query->close();
+            try {
+                $this->query->close();
+            } catch (BadMethodCallException $e) {
+                $this->query = null;
+            }
         }
```

Comment:
Ah you're right! I'm mixing things up. Sorry for that.

Reply:
I just removed the null-check because we need to keep the try/catch because during the `testDeallocatePreparedQueryThenTryToClose` method, there are two calls to `close`, and an exception is throwed

Reply:
Sorry I did not see your comments in time, and I pushed an update
The unit test are passing without the null check
Do I need to revert, or do I keep the code as it is now ?

Reply:
This is testing for BasePreparedQuery, and if there are no cases where $this->query is null, then it is fine.

Reply:
ok so I'm not touching anything anymore